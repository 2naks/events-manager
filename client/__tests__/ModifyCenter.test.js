import ConnectedModifyCenter, { ModifyCenter } from '../components/ModifyCenter.jsx';

describe('ModifyCenter Component', () => {
  const initialState = {
    centerReducer: {
      status: "Error",
      message: "",
      center:""
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => {
      return str;
    }
  }

  const props = {
    dispatch: () => {},
    show: () => {}
  };

  describe('ModifyCenter form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<ModifyCenter {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render for connected component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedModifyCenter /></Provider>);
    });

    it('it should render redirect on  success', () => {
      const wrapper = shallow(<ModifyCenter {...props} status="Success" />);
    });
  });

  describe('ModifyCenter events', () => {
    it('it should handle onChange event', () => {
      const wrapper = shallow(<ModifyCenter {...props} />);
      const descriptionInput = wrapper.find('[name="description"]');
      descriptionInput.simulate('change', { target: {} });
    });

    it('it should submit form', () => {
      const wrapper = mount(<Provider store={store}><ConnectedModifyCenter /></Provider>);
      const button = wrapper.find('.btn');
      button.simulate('click', { preventDefault: () => {} });
    });
  });
});
