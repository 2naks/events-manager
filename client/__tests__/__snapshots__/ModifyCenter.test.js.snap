// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifyCenter Component ModifyCenter form  should render self it should render for dumb component 1`] = `
<div
  className="panel add-center"
  id="add-center"
>
  <div
    className="container "
  >
    <div
      className="row"
    >
      <div
        className="container"
      >
        <form>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor=""
            />
            <div
              className="col-sm-9"
            />
          </div>
          <p
            className="text-center h5"
          >
            <b>
              MODIFY CENTER
            </b>
          </p>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Name
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="name"
                onChange={[Function]}
                type="text"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="Description"
            >
              Description
            </label>
            <div
              className="col-sm-9"
            >
              <textarea
                className="form-control"
                name="description"
                onChange={[Function]}
                type="text"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Location
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="location"
                onChange={[Function]}
                type="text"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Capacity
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                min="1"
                name="capacity"
                onChange={[Function]}
                step="1"
                type="number"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Price
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                min="1"
                name="price"
                onChange={[Function]}
                step="any"
                type="number"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Facilities
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="facilities"
                onChange={[Function]}
                type="text"
              />
              <small
                className="form-text text-muted"
                id="fileHelp"
              >
                ( Optional )
              </small>
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="available"
            >
              Availability
            </label>
            <div
              className="col-sm-9"
            >
              <select
                className="form-control"
                name="available"
                onChange={[Function]}
              >
                <option
                  value="true"
                >
                  Center is Available
                </option>
                <option
                  value="false"
                >
                  Center is not Available
                </option>
              </select>
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="exampleInputFile"
            >
              Image
            </label>
            <div
              className="col-sm-9"
            >
              <input
                aria-describedby="fileHelp"
                className="form-control"
                id="exampleInputFile"
                name="image"
                type="file"
              />
              <small
                className="form-text text-muted"
                id="fileHelp"
              >
                ( Optional )
              </small>
            </div>
          </div>
          <div
            className="form-group row"
          >
            <div
              className="offset-sm-3 col-sm-9"
            >
              <button
                className="btn btn-sm btn-success"
                onClick={[Function]}
                type="submit"
              >
                <i
                  className="fa fa-plus fa-lg"
                />
                 Update
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
